### ISSUE TITLE
Backport --pr only backports one commit even if PR merged multiple  commits into target branch 

### ISSUE THREAD

**Issue by sarod** [sqren/backport#417]
When a PR containing multiple commit is merged using 
[Rebase and merge](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github#rebasing-and-merging-your-commits) then multiple commits are created in the target branch by github.

However when backporting such PR using --pr only one of the several commit is backporting which is really confusing and error prone.

**Comment by sqren** [comment]
> when backporting such PR using --pr only one of the several commit is backporting which is really confusing and error prone.

Is there any possibility you can repro this in a dummy repo? I have some demo scenarios and I can't repro what you are seeing. For instance, https://github.com/backport-org/different-merge-strategies/commit/bdc5a17f81e5f32129e27b05c742e055c650bc54 is a merge commit. If I try to backport that using 

```
backport --repo backport-org/different-merge-strategies --pr 9 --branch 7.x
```

A PR is created with all the original commits:
https://github.com/backport-org/different-merge-strategies/pull/18/commits

**Comment by sarod** [comment]
Unfortunately the original case is on a private repository.

I'll try to setup a public repo where I can reproduce this.

**Comment by sqren** [comment]
Hi @sarod 

I was able to reproduce what you experienced in this PR: https://github.com/backport-org/backport-demo/pull/132
I see now that only the last commit is backported. This happen because multiple commits are merged onto the base branch, but no merge commit is added. 

Afaict it is not currently possible to programatically determine the resulting commits from "Rebase and merge".

> I think that backport tool should fetch the history before the merge commit and include all previous commits that are associated with the PR to the backport PR.

I'm not sure this would work reliably. When a pull request is merged, the pull request branch may be deleted. If this happens it's not possible to access the commits. Even when the branch is not deleted, someone might force push to it or modify it.
Ideally the backport tool backports commits from the base branch only (eg `main`/`master`).

I've opened a thread on this and hope someone from Github will chime in. 
https://github.community/t/commits-merged-in-via-rebase-and-merge/259570

**Comment by sarod** [comment]
@sqren from what I tested manually in my situation the merge commit is the latest commit pushed to the main branch.


 I think that the other commits can be found by:
- querying the history of the main branch right before the merge commit 
-  and using the value of associatedPullRequests field of the grphql commit object to filter only commits originiating from the backporyed PR. This property is really giving you the PR number so I think it would be kept even if the branch is deleted.

**Comment by sqren** [comment]
Hey @sarod 

I opened [a PR](https://github.com/sqren/backport/pull/422) to fix this problem. You can try it out by installing the beta version of backport:

```
yarn add backport@beta
```

You can also use the beta version of the github action:
```
uses: sqren/backport-github-action@beta
```

**Comment by sqren** [comment]
Thanks for letting me know @sarod ! 
I've released 8.9.0 which contains support for "Rebase and merge" pull requests.
